modules/aggregations/src/main/java/org/elasticsearch/aggregations/bucket/histogram/AutoDateHistogramAggregator.java

```diff
@@ -573,7 +574,11 @@ private void rebucket() {
                 long[] mergeMap = new long[Math.toIntExact(oldOrds.size())];
                 bucketOrds = new LongKeyedBucketOrds.FromMany(bigArrays());
                 success = true;
-                for (long owningBucketOrd = 0; owningBucketOrd <= oldOrds.maxOwningBucketOrd(); owningBucketOrd++) {
+                long maxOwning = oldOrds.maxOwningBucketOrd();
+                for (long owningBucketOrd = 0; owningBucketOrd <= maxOwning; owningBucketOrd++) {
+                    if (context.isCancelled()) {
```

Comment:
Nit: It is not clear to me why we are handling auto-date histogram in a special way. Could we add a comment explaining the why?

Reply:
Sure! Its just that it spends a while spinning around doing extra stuff before it's ready for the normal build.

Reply:
And I found it from an actual bug report of folks that were trying to terminate aggs and the thread was stuck here.

Reply:
Thanks

Reply:
you have not one but three typos in that sentence :D